src/main/java/org/myrobotlab/service/As5048AEncoder.java

```diff
@@ -2,18 +2,16 @@
 
 import org.myrobotlab.framework.ServiceType;
 import org.myrobotlab.logging.LoggingFactory;
-import org.myrobotlab.sensor.EncoderData;
-import org.myrobotlab.service.abstracts.AbstractEncoder;
+import org.myrobotlab.service.abstracts.AbstractPinEncoder;
 import org.myrobotlab.service.interfaces.EncoderControl;
-import org.myrobotlab.service.interfaces.EncoderController;
 
 /**
  * AS5048A - SPI based 14 bit magnetic absolute position encoder.
  * 
  * @author kwatters
  *
  */
-public class As5048AEncoder extends AbstractEncoder implements EncoderControl {
+public class As5048AEncoder extends AbstractPinEncoder implements EncoderControl {

```

Comment:
Same comment as before.  I think the base class should just be AbstractEncoder.    This is an SPI based encoder.  It's not based on a Pin.

Reply:
Same as above too ...
You can get rid of AbstractPinEncoder .. change the name to AbstractI2CEncoder or remove it completely - but previously the AbstractEncoder was not devoid of pin references ..
Now, its gone ready to be created without pin references